src/main/java/ListCommand.java

```diff
@@ -0,0 +1,28 @@
+/**
+ * Encapsulates a "list" command from the user.
+ * The "list" command takes in no arguments, and will be rejected by `Parser` if arguments are supplied.
+ */
+public class ListCommand implements Command {
+    /**
+     * Constructs a new ListCommand instance.
+     */
+    public ListCommand() {
+    }
+    
+    /**
+     * Prints the tasks in the task list to the UI.
+     */
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        for (int i = 0; i < tasks.size(); i++) {
+            ui.showNumberedEntry(i + 1, tasks.getTask(i));
```

Comment:
I read through your Command class and its child classes and I think you did a great job in making use of the Ui to print responses to the user instead of using System.out.println, which makes the code look cleaner and more elegant. Furthermore, the methods in Ui are appropriately named which makes it easy to understand. Good job!